app_test_suite/__main__.py

```diff
@@ -33,12 +34,17 @@ def get_version() -> str:
         return ver
 
 
-def get_pipeline() -> List[BuildStepsFilteringPipeline]:
-    return [
-        # FIXME: once we have more than 1 test engine, this has to be configurable
-        PytestTestFilteringPipeline(),
-    ]
-
+def get_pipeline(test_executor) -> List[BuildStepsFilteringPipeline]:
+    if test_executor == "pytest":
+        return [
+            PytestTestFilteringPipeline(),
+        ]
+    elif test_executor == "gotest":
+        return [
+            GotestTestFilteringPipeline(),
+        ]
+    else:
+        raise ConfigError("test-executor", f"Unknown executor '{test_executor}'.")
```

Comment:
We talked about detecting if there are go or python files. But I think a flag is simpler to start with and we can default to pytest to not affect existing tests. WDYT?